walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -33,7 +33,31 @@
     
     // List of TimeRanges that will not work for meeting attendees
     Collection<TimeRange> blockedTimes = new ArrayList();
+    // List of TimeRanges that will not work for meeting attendees + optional attendees
+    Collection<TimeRange> optionalBlockedTimes = new ArrayList();
+    
+    createListOfBlockedTimes(events, request, blockedTimes);
+    createListOfBlockedTimes(events, request, optionalBlockedTimes);
+
+    // List of TimeRanges that will work for meeting attendees
+    Collection<TimeRange> availableTimes = new ArrayList();
+    // List of TimeRanges that will work for meeting attendees + optional attendees
+    Collection<TimeRange> optionalAvailableTimes = new ArrayList();
+
+    createListOfAvailableTimes(events, request, blockedTimes, availableTimes);
+    createListOfAvailableTimes(events, request, optionalBlockedTimes, optionalAvailableTimes);
 
+    // Only return the List that works for optional attendees if there is one or more TimeRange
+    if (optionalAvailableTimes.size() > 0) {
+        return optionalAvailableTimes;
+    }
+    return availableTimes;
+  }
+
+  /**
+  * Create a List of TimeRanges that will not work for the given attendees based on events that are scheduled.
+  */
+  public void createListOfBlockedTimes(Collection<Event> events, MeetingRequest request, Collection<TimeRange> blockedTimes) {
     // Add the unavailable TimeRanges into the blockedTimes list
     for(String attendee: request.getAttendees()) {
         for(Event event: events){
```

Comment:
Is this ongoing? the for loop is empty

Reply:
yes! I think you might have to expand the file, this is inside of the inner for loop:
if(event.getAttendees().contains(attendee)) {
                blockedTimes.add(event.getWhen());
}